tests/cases/compiler/thisTypeInNativeThisAssignableMethods.ts

```diff
@@ -0,0 +1,55 @@
+class A {
+    options: string[];
+
+    addOptions(options: string[]) {
+        if (!this.options) {
+            this.options = [];
+        }
+        options.forEach(function (item) {
+            this.options.push(item);
+        }, this);
+        return this;
+    }
+
+    test(options: string[]) {
+        options.some(function (val, index) {
```

Comment:
I'm afraid that it would be problem. As the default compile target seems to be es5. Then it prompts that `find` not exist in `xxx[]`. Should we create a test project for it?